packages/editor/src/ColorPicker.tsx

```diff
@@ -0,0 +1,214 @@
+/** @jsx jsx */
+import { jsx } from 'theme-ui'
+import React from 'react'
+import { CustomPicker, CustomPickerProps, HSLColor } from 'react-color'
+import {
+  EditableInput,
+  Hue,
+  Saturation,
+} from 'react-color/lib/components/common'
+import { EditableInputProps } from 'react-color/lib/components/common/EditableInput'
+import { usePopoverState, Popover, PopoverDisclosure } from 'reakit/Popover'
+
+const Lens = () => (
+  <div
+    sx={{
+      width: 16,
+      height: 16,
+      borderRadius: 9999,
+      border: '1px solid white',
+      transform: 'translate(-8px, -8px)',
+    }}
+  />
+)
+
+const Handle = () => (
+  <div
+    sx={{
+      width: 12,
+      height: 12,
+      borderRadius: 9999,
+      backgroundColor: 'white',
+      boxShadow: '0 0 2px rgba(0,0,0,.25)',
+      transform: 'translate(-6px, -2px)',
+    }}
+  />
+)
+
+type InputProps = React.PropsWithoutRef<EditableInputProps> & {
+  name?: string
+}
+
+/** placeholder is used because react-color does not pass
+ * HTML attributes to the element
+ */
+const Input = (props: InputProps) => (
+  <EditableInput
+    {...props}
+    placeholder={props.name}
+    style={{
+      input: {
+        width: '100%',
+        fontSize: 12,
+        fontFamily: 'Menlo, monospace',
+        margin: 0,
+        padding: 0,
+        border: 0,
+        outline: 'none',
+      },
+      label: {
+        display: 'none',
+      },
+    }}
+  />
+)
+
+type LabelProps = React.PropsWithoutRef<JSX.IntrinsicElements['label']> & {
+  width?: string | number
+  flex?: number
+}
+
+const Label = ({ width = '100%', flex = 1, ...props }: LabelProps) => (
+  <label
+    {...props}
+    sx={{
+      display: 'block',
+      width,
+      fontSize: 10,
+      input: {
+        ':focus': {
+          color: 'primary',
+          bg: 'highlight',
+        },
+      },
+    }}
+  />
+)
+
+type CustomPickerForwardedProps = {
+  size?: number
+}
+type PickerProps = CustomPickerForwardedProps & CustomPickerProps
```

Comment:
Could we export `PickerProps` and turn it into an interface to make declaration merging possible in user's repository?

```suggestion
export interface PickerProps extends CustomPickerForwardedProps, CustomPickerProps {}
```

I agree that it looks worse, but it detects conflicts instead of resolving to `never` and it's possible to declaration merge it. https://www.typescriptlang.org/docs/handbook/declaration-merging.html#merging-interfaces

Reply:
Fixed in 14451fab.

Reply:
I just converted all `*Props` types to interfaces to make it possible to merge declarations. Maybe I misunderstood something and you just meant this one? 🤔 Let me know if I should change the others back.

Reply:
I didn't want to ask for it except here for this intersection, but it's good. Thanks!